catalog/security/catalog-security-metacardattributeplugin/src/main/java/org/codice/ddf/catalog/security/policy/metacard/MetacardAttributeSecurityPolicyPlugin.java

```diff
@@ -0,0 +1,127 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ **/
+package org.codice.ddf.catalog.security.policy.metacard;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.Result;
+import ddf.catalog.operation.Query;
+import ddf.catalog.operation.ResourceRequest;
+import ddf.catalog.operation.ResourceResponse;
+import ddf.catalog.plugin.PolicyPlugin;
+import ddf.catalog.plugin.PolicyResponse;
+import ddf.catalog.plugin.StopProcessingException;
+import ddf.catalog.plugin.impl.PolicyResponseImpl;
+
+/**
+ * Plugin that parses Metacard attributes for security policy information
+ */
+public class MetacardAttributeSecurityPolicyPlugin implements PolicyPlugin {
+
+    List<String> metacardAttributes = new ArrayList<>();
+
+    public List<String> getMetacardAttributes() {
+        return metacardAttributes;
+    }
+
+    public void setMetacardAttributes(List<String> metacardAttributes) {
+        this.metacardAttributes = metacardAttributes;
+    }
+
+    private Map<String, Set<String>> buildSecurityMap(Metacard metacard) {
+        Map<String, Set<String>> securityMap = new HashMap<>();
+        if (metacard != null) {
+            for (String metacardAttribute : metacardAttributes) {
+                Attribute attribute = metacard.getAttribute(metacardAttribute);
+                if (attribute != null) {
+                    securityMap.put(metacardAttribute,
+                            new HashSet<>(listAsStrings(attribute.getValues())));
+                }
+            }
+        }
+        return securityMap;
+    }
+
+    private List<String> listAsStrings(List list) {
+        return list;
```

Comment:
This method seems a little odd. I ran into a similar issue the other day with converting the attribute list of serializable objects into a list of strings and ended up with this,
```return new HashSet<>(getAttribute(TAGS).getValues()
                    .stream()
                    .map(String::valueOf)
                    .collect(Collectors.toList()));```
Not as efficient as you way but maybe more straight forward? I see that this particular case maybe coming up frequently and we might want to figure out what the best way to handle it is and then consistently use it.